File: Arcade/Intro/Ruby/deleteDigit.rb
def deleteDigit(n)
    max, s = 0, n.to_s
    s.size.times do |i|
        v = (0..s.size).to_a.filter{|j| j != i}.map{|j| s[j]}.join.to_i
        max = v if v > max
    end
    return max
end

File: Arcade/Intro/javascript/deleteDigit.js
function deleteDigit(n) {
    let max = 0, idx = [], s = (n+"");
    
    for(let i = 0; i < s.length; i++) idx.push(i);
    for(let i = 0; i < s.length; i++) {
        let v = parseInt(idx.filter(j => j != i).map(j => s[j]).join(""));
        if(v > max) max = v;
    }
    
    return max;
}